### DISCUSSION TITLE
How to handle an extractor error and give a custom reponse

### DISCUSSION THREAD

**Discussion by edg-l** [tokio-rs/axum#1451] [Q&A] [answered]
For example, when using a Json extractor to deserialize a body, if the body is incorrect, it responds with a 422 and a string body like: Failed to deserialize the JSON body into the target type: missing field `password` at line 4 column 1

What I want to know is how to handle this and give for example a JSON response.

**Comment by jplatte** [comment, marked as answer]
See the [customize-extractor-error](https://github.com/tokio-rs/axum/tree/a7d89541786167e990d095a03ac7bdba68d7a55a/examples/customize-extractor-error) example.
